pkg/test/echo/server/instance.go

```diff
@@ -275,7 +275,9 @@ func (s *Instance) startMetricsServer() {
 		log.Errorf("could not set up prometheus exporter: %v", err)
 		return
 	}
-	mux.Handle("/metrics", LogRequests(exporter))
+	if exporter != nil {
```

Comment:
I don't really get how this (generally the PR, not this specific line) is supposed to work. So a vendor wants to fork -- no problem. But how is this supposed to be maintained, and why should it be maintained in OSS instead of in the vendor's fork?

Having an `if` that can never be `false` is a code smell -- it will probably be "fixed" in a cleanup later, and a linter may even catch it and block it (not sure if its that smart).

How should we know not to move the `prometheus.go` back under `monitoring.go`? Etc

This seems like the vendors that want to make these types of changes should have smarter fork management, rather than pushing fragile requirements into OSS.

And to be clear --  I am very much in favor of making parts of Istio pluggable. But it needs clear and meaningful contracts

Reply:
I think the strategy is clear - move Prometheus Otel alpha code away into its own code unit. We can create a package as well. I think it's up to Istio to decide how to do dependency injection, but using an unstable code in core `pkg/monitoring` is a code smell.

Reply:
Istio is not a library or meant to be consumed by anything other than the Istio project to build the code and test as part of Istio. Just like https://github.com/kubernetes/kubernetes/. The module structure and dependencies is designed to reflect that.

I would agree having a "core" package depend on an "alpha" package would be a mistake in a library offering such semantics, but that is not the goal of the Istio repo.

Reply:
Sure, no strong feelings - we can manage a fork, I was only trying to upstream improvements.